oauth2client/client.py

```diff
@@ -1222,6 +1222,43 @@ def create_scoped(self, scopes):
         """
         return self
 
+    @classmethod
+    def from_json(cls, s):
+        from oauth2client.service_account import _ServiceAccountCredentials
+        s = _from_bytes(s)
+        data = json.loads(s)
+
+        # A GoogleCredentials.from_stream return value might be a
+        # _ServiceAccountCredentials instance, so we should handle it in from_json.
+        if (data['_module'] == 'oauth2client.service_account' and
+            data['_class'] == '_ServiceAccountCredentials'):
+            return _ServiceAccountCredentials.from_json(s)
+
+        if (data.get('token_expiry') and
```

Comment:
Would probably be better to just use

```python
expiry = data.get('token_expiry')
```

and then refer to `expiry` instead of `data['token_expiry']` after (twice).